next/src/utils/sanity-client.ts

```diff
@@ -0,0 +1,40 @@
+import { createClient as createSanityClient } from 'next-sanity';
+import type { SanityClient, QueryParams } from '@sanity/client';
+
+interface CreateClientConfig {
+  projectId: string;
+  dataset: string;
+  apiVersion: string;
+  useCdn: boolean;
+}
+
+type CreateClient = (config: CreateClientConfig) => SanityClient;
+
+const createClient = createSanityClient as CreateClient;
+
+const projectId = process.env.SANITY_PROJECT_ID || '';
+const dataset = process.env.SANITY_DATASET || '';
+const apiVersion = process.env.SANITY_API_VERSION || 'v2022-03-07';
```

Comment:
Change apiVersion fallback to newest date (2024-01-16). I would rather not use env here.